app/src/main/java/com/babylon/wallet/android/presentation/onboarding/restore/mnemonics/RestoreMnemonicsViewModel.kt

```diff
@@ -277,6 +278,12 @@ class RestoreMnemonicsViewModel @Inject constructor(
         val isMainBabylonSeedPhrase: Boolean
             get() = recoverableFactorSource?.factorSource?.id == mainBabylonFactorSourceId
 
+        val showAdvancedMode: Boolean
+            get() = recoverableFactorSource?.factorSource?.hasOlympiaSeedPhraseLength ?: false
```

Comment:
Are you sure this is the correct way of doing it? `hasOlympiaSeedPhraseLength` is 24 similar to babylon. This is not the correct way of looking if a seed phrase belongs to an account created with babylon. I think a better approach is `FactorSourceCryptoParameters.supportsOlympia`.

Reply:
if we would add same seed phrase in account recovery scan both as babylon and olympia, of size 24, you would have a seed that supports both olympia and babylon and length is 24

Reply:
`hasOlympiaSeedPhraseLength` checks if the word count is different than 24 and I thought this is enough to know it's not a Babylon seed phrase. But it's good to know there is a better and more reliable way to check this. Thank you!